### ISSUE TITLE
False JavaScript Compile-Time Error

### ISSUE THREAD

**Issue by erosman** [mozilla/amo-validator#442]
Example:

`https://addons.mozilla.org/en-US/firefox/files/browse/459606/file/modules/BrandThunderUtils.jsm#top`

> JavaScript Compile-Time Error
> 
> A compile-time error in the JavaScript halted validation of that file.
> 
> Message: missing ] after element list
> modules/BrandThunderUtils.jsm
> 8
> 9
> 10
> Cu.import("resource://gre/modules/AddonManager.jsm");
> try {
>   Components.utils.import("resource:///modules/NewTabURL.jsm");

Actual code:

```
var EXPORTED_SYMBOLS = ["BrandThunderUtils"];

const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
Cu.import("resource://gre/modules/XPCOMUtils.jsm");
Cu.import("resource://gre/modules/Services.jsm");
Cu.import("resource://gre/modules/NetUtil.jsm");
Cu.import("resource://gre/modules/LightweightThemeManager.jsm");
Cu.import("resource://gre/modules/AddonManager.jsm");
try {
  Components.utils.import("resource:///modules/NewTabURL.jsm");
} catch (e) {}
```

**Comment by wagnerand** [comment]
For some reason, we always report line 9 as being invalid JS. In this case, the real error is
`var s = [toHexString(hash.charCodeAt(i)) for (i in hash)].join('');`
(line 370)

This line is using array comprehension which were initially part of ES6, but removed along the way and therefore removed from Spidermonkey as well. I don't remember the exact version, but I think it was between 45 and 48.

Long story short: This file is not valid JavaScript (as per ES6) and Spidermonkey (which is used by the validator is not wrong).
